cmd/exchange_template/wrapper_file.tmpl

```diff
@@ -465,9 +513,83 @@ func ({{.Variable}} *{{.CapitalName}}) GetLatestFundingRates(_ context.Context,
 	return nil, common.ErrNotYetImplemented
 }
 
+// GetHistoricalFundingRates returns funding rates for a given asset and currency for a time period
+func ({{.Variable}} *{{.CapitalName}}) GetHistoricalFundingRates(_ context.Context, r *fundingrate.HistoricalRatesRequest) (*fundingrate.HistoricalRates, error) {
+	if r == nil {
+		return nil, common.ErrNilPointer
+	}
+	return nil, common.ErrNotYetImplemented
+}
+
+// GetOpenInterest returns the open interest rate for a given asset pair
+func ({{.Variable}} *{{.CapitalName}}) GetOpenInterest(_ context.Context, _ ...key.PairAsset) ([]futures.OpenInterest, error) {
+	return nil, common.ErrNotYetImplemented
+}
+
+// GetCurrencyTradeURL returns the URL to the exchange's trade page for the given asset and currency pair
+func ({{.Variable}} *{{.CapitalName}}) GetCurrencyTradeURL(_ context.Context, a asset.Item, cp currency.Pair) (string, error) {
+	_, err := {{.Variable}}.CurrencyPairs.IsPairEnabled(cp, a)
+	if err != nil {
+		return "", err
+	}
+	return "", common.ErrNotYetImplemented
+}
+
 // UpdateOrderExecutionLimits updates order execution limits
 func ({{.Variable}} *{{.CapitalName}}) UpdateOrderExecutionLimits(_ context.Context, _ asset.Item) error {
 	return common.ErrNotYetImplemented
 }
 
+// SetLeverage sets the account's initial leverage for the asset type and pair
+func ({{.Variable}} *{{.CapitalName}}) SetLeverage(_ context.Context, _ asset.Item, _ currency.Pair, _ margin.Type, _ float64, _ order.Side) error {
```

Comment:
Can you please do a check for these wrapper functions whether they are `exchange.Base` methods. We can remove them as methods and these will need to be forcefully applied to all exchanges for standardisation.